File: medplum-java/medplum-model/src/main/java/com/fhir/CapabilityStatement_Software.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableCapabilityStatement_Software.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface CapabilityStatement_Software {
  Optional<String> version();

  Optional<List<Extension>> modifierExtension();

  Optional<List<Extension>> extension();

  Optional<String> name();

  Optional<String> id();

  Optional<dateTime> releaseDate();

  static ImmutableCapabilityStatement_Software.Builder builder() {
    return ImmutableCapabilityStatement_Software.builder();
  }
}

File: medplum-typescript/src/model/fhir/CapabilityStatement_Software.ts
import { dateTime } from "./dateTime";
import { Extension } from "./Extension";

export interface CapabilityStatement_Software {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    name: string | null | undefined;
    version: string | null | undefined;
    releaseDate: dateTime | null | undefined;
}
